src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,214 @@
+  if (!category) {
+    throw new Error('No category in collection');
+  }
+  return category;
+}
+
+export function getGeographicDescription(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:geographic_description'] as string;
+}
+
+export function getEvent(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:event_name'] as string;
+}
+
+export function getRegion(collection: StacCollection): string {
+  const region = collection['linz:region'] as string;
+  if (!region) {
+    throw new Error('No region in collection');
+  } else if (!regions.includes(region)) {
+    throw new Error(`Invalid region: ${region}`);
+  }
+  return region;
+}
+
+export function getDate(collection: StacCollection): string {
+  const interval = collection.extent.temporal.interval[0];
+  const startYear = interval[0]?.slice(0, 4);
+  const endYear = interval[1]?.slice(0, 4);
+
+  if (!startYear || !endYear) {
+    throw new Error(`Missing datetime in interval: ${interval}`);
+  }
+  if (startYear === endYear) {
+    return startYear;
+  }
+  return `${startYear}-${endYear}`;
+}
+
+/*
+ *  nb: The following functions: 'loadFirstTiff', 'extractGsd', and 'extractEpsg' are workarounds
+ *  to manage the block information from being added to the collection (eo stac extension).
+ *  Once this is fixed the following functions can be replaced.
+ */
+
+/**
+ * Gets first item & tiff listed in collection
+ *
+ * @async
+ * @param {string} source
+ * @param {StacCollection} collection
+ * @returns {Promise<CogTiff>}
+ */
+export async function loadFirstTiff(source: string, collection: StacCollection): Promise<CogTiff> {
+  const itemLink = collection.links[2]?.href.replace('./', ''); // [2] to skip root & self links
+  if (itemLink == null) {
+    throw new Error(`No items in collection from: ${source}`);
+  }
+  const itemPath = fsa.join(source, itemLink);
+  const item = await fsa.readJson<StacItem>(itemPath);
+  if (item == null) throw new Error(`Failed to get item.json from ${itemPath}.`);
+  const tiffLink = item.assets['visual']?.href.replace('./', '');
+  if (tiffLink == null) {
+    throw new Error(`No tiff assets in Item: ${itemPath}`);
+  }
+  const tiffPath = fsa.join(source, tiffLink);
+  const tiff = await createTiff(tiffPath);
+  if (tiff == null) throw new Error(`Failed to get tiff from ${tiffPath}.`);
+  return tiff;
+}
+
+export function extractGsd(tiff: CogTiff): string {
+  const gsd = tiff.images[0]?.resolution[0];
+  if (!gsd) {
+    throw new Error(`Missing resolution tiff tag`);
+  }
+  return `${gsd}m`;
+}
+
+export function extractEpsg(tiff: CogTiff): string {
+  const epsg = tiff.images[0]?.epsg;
+  if (!epsg) {
+    throw new Error(`Missing epsg tiff tag`);
+  } else if (!Epsg.Codes.has(epsg)) {
+    throw new Error(`Invalid EPSG code: ${epsg}`);
+  }
+  return `${epsg}`;
```

Comment:
this should really kept as a number until its ready to be serialized into a path.